src/pages/[collectionSlug]/page/[page].astro

```diff
@@ -0,0 +1,115 @@
+---
+import CollectionItemList from "@/components/CollectionItemList.astro";
+import PagesSection from "@/components/PagesSection.astro";
+import Layout from "@/layouts/Layout.astro";
+import PaginationNav from "@/components/PaginationNav.astro";
+import {
+  fetchCustomCollectionPages,
+  createCustomCollectionPageMapper,
+  fetchCollection,
+  fetchSlug,
+} from "@/utilities/fetch";
+import { paginate } from "@/utilities/pagination";
+import { safeJsonParse } from "@/utilities/fetch/payload-fetch";
+
+export const PAGE_SIZE = import.meta.env.PAGE_SIZE || 10;
+const { collectionSlug, page } = Astro.params;
+const currentPage = parseInt(page) || 1;
+
+// Check if this slug matches a custom collection config
+const collectionConfig = await fetchSlug(
+  "custom-collections",
+  collectionSlug || "",
+);
+
+if (!collectionConfig) {
+  return Astro.redirect("/404");
+}
+
+// Fetch pages for this custom collection
+const collectionPagesData = await fetchCustomCollectionPages(
+  collectionConfig.id,
+);
+const collectionPages = collectionPagesData?.docs || [];
+
+// Sort by publishedAt
+const sorted = collectionPages.sort((a: any, b: any) => {
+  return (
+    new Date(b.publishedAt || 0).getTime() -
+    new Date(a.publishedAt || 0).getTime()
+  );
+});
+
+// Paginate
+const hasPaginationNav = sorted.length >= PAGE_SIZE;
+const { totalPages, paginatedItems } = paginate(sorted, currentPage, PAGE_SIZE);
+
+// Map items
+const mapper = createCustomCollectionPageMapper(collectionSlug || "");
+const items = paginatedItems.map(mapper);
+
+export async function getStaticPaths() {
+  // Fetch all custom collection configs to generate static paths
+  try {
+    const response = await fetchCollection("custom-collections");
+    const data = await safeJsonParse(response);
+    const configs = data?.docs || [];
+
+    const paths: any[] = [];
+
+    for (const config of configs) {
+      const pagesResponse = await fetchCollection(
+        `custom-collection-pages?where[collectionConfig][equals]=${config.id}&limit=0`,
+      );
+      const pagesData = await safeJsonParse(pagesResponse);
```

Comment:
Delete and rename the `pagesResponse` above to `pagesData` since it already gets called in the `fetchCollection`